azure-ipam/go.mod

```diff
@@ -1,11 +1,12 @@
 module github.com/Azure/azure-container-networking/azure-ipam
 
 go 1.22.7
+toolchain go1.23.3
```

Comment:
Hmm, Dependabot added this automatically... May want to keep an eye on this as we go forward. 

@rbtr for visibility

Reply:
not an immediate concern - we use MS Go build containers which is built to ignore `toolchain`